src/Evolution/DiscontinuousGalerkin/MortarData.hpp

```diff
@@ -69,6 +75,68 @@ class MortarData {
       std::vector<double> neighbor_mortar_vars) noexcept;
   //@}
 
+  /*!
+   * \brief Insert the magnitude of the local face normal, the determinant
+   * of volume inverse Jacobian, and determinant of the face Jacobian. Used
+   * for local time stepping with Gauss points.
+   *
+   * The magnitude of the face normal is given by:
+   *
+   * \f{align*}{
+   *  \sqrt{
+   *   \frac{\partial\xi}{\partial x^i} \gamma^{ij}
+   *   \frac{\partial\xi}{\partial x^j}}
+   * \f}
+   *
+   * for a face in the \f$\xi\f$-direction, with inverse spatial metric
+   * \f$\gamma^{ij}\f$.
+   */
+  void insert_local_geometric_quantities(
+      const Scalar<DataVector>& local_volume_det_inv_jacobian,
+      const Scalar<DataVector>& local_face_det_jacobian,
+      const Scalar<DataVector>& local_face_normal_magnitude) noexcept;
+
+  /*!
+   * \brief Insert the magnitude of the local face normal. Used for local time
+   * stepping with Gauss-Lobatto points.
+   *
+   * The magnitude of the face normal is given by:
+   *
+   * \f{align*}{
+   *  \sqrt{
+   *   \frac{\partial\xi}{\partial x^i} \gamma^{ij}
+   *   \frac{\partial\xi}{\partial x^j}}
+   * \f}
+   *
+   * for a face in the \f$\xi\f$-direction, with inverse spatial metric
+   * \f$\gamma^{ij}\f$.
+   */
+  void insert_local_face_normal_magnitude(
+      const Scalar<DataVector>& local_face_normal_magnitude) noexcept;
+
+  /*!
+   * \brief Sets the `local_volume_det_inv_jacobian` by setting the DataVector
+   * to point into the correct memory. The result should never be changed.
```

Comment:
I agree. I'm not super worried about people misusing this class, mostly because it's nested so deeply in the algorithm that, if you're using this, you've had to understand pretty much all the functions in `ComputeTimeDerivative`, including the helpers. The easiest way of making this safe that I can think of is to store three scalars internally, and them write custom copy (and maybe move) semantics to get those right.

Reply:
If we did want to fix it, I agree your proposal sounds reasonable... but if you're not worried then I'm not worried, so let's not do this now.